### PR TITLE
Feature/54 etl cron job

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [EQ-54](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-54)

## Main Changes:
* Set up a cron job for the ATTAINS ETL process.
  * For testing, this is set to run every 30 minutes.
* Created a table to log the process.
* Created a table to store current schemas.
* Set up logic to rotate the new schema into use after creation.
* Set up logic to remove old schemas.
  * For testing, this is executed on schemas older than one day.

## Steps To Test:
1. In the `database/docker/postgres` directory, start the test database with the command `docker compose up`
2. In the `etl` directory, start the ETL process with the command `npm run start`
3. Navigate to [localhost:8080](http://localhost:8080)
4.  Login with credentials "PostgreSQL", "postgres", "postgres", "example", "expert_query" for the System, Server, Username, Password, & Database fields, respectively
5. Confirm that a "logging" schema is created with the tables "etl_log" and "etl_schemas", and that the tables are populated with current values
6. Logout, and log back in with credentials "PostgreSQL", "postgres", "eq", "expert", "expert_query" for the System, Server, Username, Password, & Database fields, respectively
7. Confirm that a schema is created with the label "schema_{timestamp}", and that it is populated with a placeholder table
8. Check that logs are being output properly to the terminal
9. Check the logs to confirm that the process is being run every 30 minutes

### CODE REVIEW

File: etl/app/server/database.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
+import pg from 'pg';
+
+import { logger as log } from './utilities/logger.js';
+import * as create from './queries/create.js';
+
+const { Pool } = pg;
+
+const dbName = 'expert_query';
```
Comment (cschwinderg) [inline etl/app/server/database.js:8]:
```suggestion
const dbName = process.env.DB_NAME ?? 'expert-query';
```

Switch this to pull from `.env`.

**Review by cschwinderg — CHANGES_REQUESTED** [review]
Works well! I manually updated the rate at which the cron runs on my local, so I could test more quickly and it cycled through the schemas as expected.

I listed some potential improvements below.

File: etl/app/server/database.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  }
+
+  try {
+    try {
+      // Create the expert_query db and ignore any errors, i.e. the db already exists
+      await client.query('CREATE DATABASE ' + dbName);
+      log.info(`${dbName} database created!`);
+    } catch (err) {
+      log.info(`Warning: ${dbName} database! ${err}`);
+    }
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply etl/app/server/database.js:77]: Sounds good, I agree, the warning is a bit confusing.

File: etl/package.json [hunk 3]
```diff
@@ -1,71 +1,72 @@
...
+    "dotenv": "16.0.3",
+    "express": "4.18.2",
+    "fast-csv": "4.3.6",
+    "jsforce": "1.11.0",
+    "log4js": "6.7.0",
+    "node-cron": "3.0.2",
+    "numeral": "2.0.6",
+    "pg": "8.8.0",
+    "us-state-codes": "1.1.2"
+  },
```
Comment (cschwinderg) [inline etl/package.json:48]:
```suggestion
  "dependencies": {
    "aws-sdk": "2.1232.0",
    "dotenv": "16.0.3",
    "express": "4.18.2",
    "log4js": "6.7.0",
    "node-cron": "3.0.2",
    "pg": "8.8.0"
  },
```

When I created this package.json file, I didn't spend very much time looking through the packages that were in there.

Packages to remove:
* canada - This is used fetching cities/provinces in Canada. We won't need this in the future.
* change-case - We probably won't need this. We can add this back in later if necessary.
* fast-csv - I think we will need a csv library, but I want to do some research as there may be a better library. This one doesn't seem to have much usage or support.
* jsforce - This is a Salesforce library, which we don't need.
* numeral - We probably won't need this. We can add this back in later if necessary.
* us-state-codes - We won't need this. If we do need to get state codes, we will just fetch them from attains (https://attains.epa.gov/attains-public/api/states) like we do in HMW.

**Review by cschwinderg — APPROVED** [review]
Thanks for making those changes.

I'm going to merge this in and then start working on the ETL build process. Once I confirm everything works in cloud.gov, I'll switch the cron job to run every 24 hours.

**Comment by maxdiebold-erg** [comment]
@cschwinderg Sounds good! I missed getting those vcap environment variables set up, though. We need that for cloud.gov, right?

**Comment by cschwinderg** [comment]
@maxdiebold-erg I'll get those as I work on the etl build/deployment process.

**Pull request merged**